netdiff/parsers/wireguard.py

```diff
@@ -0,0 +1,108 @@
+from copy import deepcopy
+from datetime import datetime, timedelta
+
+from ..exceptions import ConversionException, ParserError
+from .base import BaseParser
+
+
+class WireguardParser(BaseParser):
+    """wireguard parser"""
+
+    protocol = 'Wireguard Status Log'
+    version = '1'
+    metric = 'static'
+    max_time_diff = timedelta(minutes=5)
+
+    def to_python(self, data):
+        try:
+            return super().to_python(data)
+        except ConversionException as e:
+            return self._wg_dump_to_python(e.data)
+
+    def __parse_value(self, value):
+        return None if value == '(none)' else value
+
+    def _wg_dump_to_python(self, data):
+        lines = map(lambda line: line.split('\t'), data.strip().split('\n'))
+        try:
+            parsed_lines = self._parse_lines(lines)
+        except ValueError:
+            raise ParserError('Unrecognized format')
+        return parsed_lines
+
+    def _parse_lines(self, lines):
+        parsed_lines = {}
+        last_device = None
+        for device, *options in lines:
+            if device != last_device:
+                last_device = device
+                _, public_key, listen_port, fwmark = map(self.__parse_value, options)
+                parsed_lines.setdefault(
+                    device,
+                    {
+                        'publicKey': public_key,
+                        'listenPort': listen_port,
+                        'fwmark': fwmark,
+                        'peers': [],
+                    },
+                )
+            else:
+                (
+                    public_key,
+                    preshared_key,
+                    endpoint,
+                    allowed_ips,
+                    latest_handshake,
+                    transfer_rx,
+                    transfer_tx,
+                    persistent_keepalive,
+                ) = map(self.__parse_value, options)
+                connected, latest_handshake = self._parse_latest_handshake(
+                    latest_handshake
+                )
+                parsed_lines[device]['peers'].append(
+                    {
+                        public_key: {
+                            'presharedKey': preshared_key,
+                            'endpoint': endpoint,
+                            'latestHandshake': latest_handshake.strftime(
+                                '%Y-%m-%dT%H:%M:%SZ'
+                            ),
+                            'transferRx': transfer_rx,
+                            'transferTx': transfer_tx,
+                            'persistentKeepalive': persistent_keepalive,
+                            'allowedIps': allowed_ips.split(','),
```

Comment:
let's convert these properties from camelCase to lower case and underscores as if it's python code, this will also improve readability in the UI because the underscore is rendered as space.